File: src/main/java/ch/junamin/controllers/SolverController.java
package ch.junamin.controllers;

import java.util.List;

import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import ch.junamin.models.Board;
import ch.junamin.services.Solver;

@RestController
@RequestMapping("/api")
@SpringBootApplication
public class SolverController {

    @PostMapping("/solve")
    public ResponseEntity<String> solveBimaru(@RequestBody String requestBody) {
        try {
            Board board = new Board(requestBody);
            Solver solver = new Solver(board);
            solver.solve();
            List<String> solutions = solver.getUniqueSolutions();
            String response = String.join("\n\n", solutions);
            return ResponseEntity.ok(response);
        } catch (Exception e) {
            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Fehler beim Verarbeiten der Anfrage.");
        }
    }
}

File: dotnet-core/BimaruApi/Controllers/SolverController.cs
using BimaruApi.Domain;
using Microsoft.AspNetCore.Mvc;

namespace BimaruApi.Controllers
{
    [Route("api")]
    [ApiController]
    public class SolverController : ControllerBase
    {
        [HttpGet("home")]
        public IActionResult Home() 
        {
            return Ok("Hello Bimaru Universe of dotnet-core!");
        }

        [HttpPost("solve")]
        [Consumes("text/plain")]
        public async Task<IActionResult> SolveBimaru()
        {
            try
            {
                using var reader = new StreamReader(Request.Body);
                var requestBody = await reader.ReadToEndAsync();

                if (string.IsNullOrWhiteSpace(requestBody))
                {
                    return BadRequest("Leerer Request-Body ist nicht erlaubt.");
                }
                
                Board board = new(requestBody);
                Solver solver = new(board);
                solver.Solve();
                var solutions = solver.GetUniqueSolutions();

                return Ok(solutions.Count > 0
                    ? string.Join("\n\n", solutions) 
                    : "Keine Lösung gefunden.");
            }
            catch (Exception ex)
            {
                Console.WriteLine($"Fehler beim Verarbeiten der Anfrage: {ex.Message}");
                return BadRequest("Fehler beim Verarbeiten der Anfrage.");
            }
        }
    }
}
